test/unit/hibf/interleaved_bloom_filter_test.cpp

```diff
@@ -96,6 +96,34 @@ TEST(ibf_test, construction_from_config)
     EXPECT_RANGE_EQ(agent.bulk_contains(0), expected_v0);
 }
 
+TEST(ibf_test, construction_from_config_with_max_bin_elements)
+{
+    std::vector<std::vector<size_t>> hashes{{1u, 2u, 3u, 4u, 5u, 6u, 7u, 8u, 9u, 10u}, {0u, 2u, 3u, 4u, 5u}};
+    size_t const number_of_ub{hashes.size()};
+
+    seqan::hibf::config ibf_config{.input_fn =
+                                       [&](size_t const num, seqan::hibf::insert_iterator it)
+                                   {
+                                       for (auto const hash : hashes[num])
+                                           it = hash;
+                                   },
+                                   .number_of_user_bins = number_of_ub};
+
+    seqan::hibf::interleaved_bloom_filter ibf_with_config{ibf_config};
+    seqan::hibf::interleaved_bloom_filter ibf_with_config_and_0_elements{ibf_config, 0u};
+    seqan::hibf::interleaved_bloom_filter ibf_with_config_and_10_elements{ibf_config, 10u};
+    seqan::hibf::interleaved_bloom_filter ibf_with_config_and_20_elements{ibf_config, 20u};
```

Comment:
```suggestion
    seqan::hibf::interleaved_bloom_filter only_config{ibf_config};
    seqan::hibf::interleaved_bloom_filter default_num_elements{ibf_config, 0u};
    seqan::hibf::interleaved_bloom_filter appropriate_num_elements{ibf_config, 10u};
    seqan::hibf::interleaved_bloom_filter larger_num_elements{ibf_config, 20u};
```

It took me a few minutes to understand what is what and what to expect. renaming those would help I think.